### PR TITLE
Use the AWS Java SDK directly for proxy support

### PR DESCRIPTION
**Pull request by michaelb990**
Fixes a bug that I didn't realize until we deployed. The AWScala wrapper around the Java SDK for AWS isn't as mature as I'd hoped and has no place to add proxies. This changes to simply using the official SDK from AWS (in Java).

### CODE REVIEW

File: src/main/scala/com/socrata/decima/util/S3AccessFromConfig.scala [hunk 1]
```diff
@@ -1,12 +1,20 @@
 package com.socrata.decima.util
 
-import awscala._
-import awscala.s3._
+import com.amazonaws.ClientConfiguration
+import com.amazonaws.auth.BasicAWSCredentials
+import com.amazonaws.services.s3.AmazonS3Client
 
 object S3AccessFromConfig {
```
Comment (sokrahta) [inline src/main/scala/com/socrata/decima/util/S3AccessFromConfig.scala]: this should be named `AmazonS3ClientFromConfig` or similar

File: src/main/scala/com/socrata/decima/data_access/S3Access.scala [hunk 2]
```diff
@@ -39,15 +49,16 @@ class S3Access(s3: S3, bucketName: String) extends S3AccessBase with Logging {
...
-    val buildInfoObject = s3.get(bucket.get, fileKey)
-    if (buildInfoObject.isEmpty) throw new RuntimeException(s"Unable to find the build info at $s3Url")
     try {
-      val buildInfo = yaml.load(buildInfoObject.get.content).asInstanceOf[S3BuildInfo]
-      buildInfo.s3Url = s3Url
+      val buildInfoObject = s3.getObject(bucketName, fileKey)
+      val inputStream = buildInfoObject.getObjectContent
+      val buildInfo = yaml.load(inputStream).asInstanceOf[S3BuildInfo]
+      buildInfoObject.close()
+    buildInfo.s3Url = s3Url
```
Comment (sokrahta) [inline src/main/scala/com/socrata/decima/data_access/S3Access.scala]: indentation

File: src/main/scala/com/socrata/decima/data_access/S3Access.scala [hunk 3]
```diff
@@ -24,9 +26,17 @@ class S3Access(s3: S3, bucketName: String) extends S3AccessBase with Logging {
...
-    if (builds.isEmpty) throw new RuntimeException(s"Unable to find any builds under $project")
-    builds.map(_.left.get).toList
+    var objectListing = s3.listObjects(new ListObjectsRequest()
+                                            .withBucketName(bucketName)
+                                            .withPrefix(s"$project/")
+                                            .withDelimiter("/"))
+    var prefixes = objectListing.getCommonPrefixes.asScala.toList
+    while(objectListing.isTruncated) {
+      objectListing = s3.listNextBatchOfObjects(objectListing)
+      prefixes ++= objectListing.getCommonPrefixes.asScala.toList
```
Comment (sokrahta) [inline src/main/scala/com/socrata/decima/data_access/S3Access.scala]: this get prefixes code is used twice and could be extracted into a local function

File: src/main/scala/com/socrata/decima/util/S3AccessFromConfig.scala [hunk 4]
```diff
@@ -1,12 +1,20 @@
...
-  def apply(s3Config: S3Config): S3 = {
-    implicit val region = Region.US_WEST_2
-    S3(s3Config.accessKeyId,
-        s3Config.secretAccessKey)
+  def apply(s3Config: S3Config): AmazonS3Client = {
+    val credentials = new BasicAWSCredentials(s3Config.accessKeyId, s3Config.secretAccessKey)
+    if(sys.env.contains("http_proxy")) {
+      val proxyArgs = sys.env("http_proxy").replace("http://", "").split(":")
+      val proxyHost = proxyArgs.head
+      val proxyPort = proxyArgs.last.toInt
```
Comment (sokrahta) [inline src/main/scala/com/socrata/decima/util/S3AccessFromConfig.scala]: what if `proxyArgs` doesn't have two members?

what if the url included `https://`?

Comment (michaelb990) [inline src/main/scala/com/socrata/decima/util/S3AccessFromConfig.scala]: This is assuming that we are passing the default `http_proxy` string for the environment. Do you have a preference on stronger error checking on the environment variable or pulling this out into the config file?

File: src/main/scala/com/socrata/decima/data_access/S3Access.scala [hunk 5]
```diff
@@ -39,15 +53,16 @@ class S3Access(s3: S3, bucketName: String) extends S3AccessBase with Logging {
...
+      val buildInfoObject = s3.getObject(bucketName, fileKey)
+      val inputStream = buildInfoObject.getObjectContent
+      val buildInfo = yaml.load(inputStream).asInstanceOf[S3BuildInfo]
+      buildInfoObject.close()
       buildInfo.s3Url = s3Url
       buildInfo
     } catch {
       case e: Exception =>
-        throw new RuntimeException("Unable to parse build_info.yml file (it may be incorrectly formatted):\n"
+        throw new RuntimeException(s"Unable to download and parse build_info.yml at $s3Url. Is it present?\n"
```
Comment (mhotan) [inline src/main/scala/com/socrata/decima/data_access/S3Access.scala]: You wrap an exception explicitly within another Exception by including it in the constructor.  
IE. `new RuntimeException("Your message", e)`

File: src/main/scala/com/socrata/decima/util/AutoprodUtils.scala [hunk 6]
```diff
@@ -7,7 +7,7 @@ import org.json4s.jackson.JsonMethods._
 
 object AutoprodUtils extends Logging {
 
-  def getLatestBuild(buildIds: List[String]): String = {
+  def getLatestBuild(buildIds: Seq[String]): String = {
```
Comment (mhotan) [inline src/main/scala/com/socrata/decima/util/AutoprodUtils.scala]: Nice!

**Comment by mhotan** [comment]
Feel free to change that exception wrapping line but other then that looks good.

**Comment by mhotan** [comment]
:shipit: :two_men_holding_hands:

**Pull request merged**